sklearn/covariance/graph_lasso_.py

```diff
@@ -702,8 +704,276 @@ def fit(self, X, y=None):
+    GraphLasso, GraphLassoCV
+
+    Notes
+    -----
+    The algorithm employed to solve this problem is the GLasso algorithm,
+    from the Friedman 2008 Biostatistics paper. It is the same algorithm
+    as in the R `glasso` package.
+
+    One possible difference with the `glasso` R package is that the
+    diagonal coefficients are not penalized.
+
+    """
+    return graphical_lasso(emp_cov, alpha, cov_init, mode, tol,
+                           enet_tol, max_iter, verbose, return_costs,
+                           eps, return_n_iter)
+
+
+@deprecated("The 'GraphLasso' was renamed to 'GraphicalLasso' "
+            "in version 0.20 and will be removed in 0.22.")
+class GraphLasso(GraphicalLasso):
+    """Sparse inverse covariance estimation with an l1-penalized estimator.
+
+    This class implements the Graphical Lasso algorithm.
+
+    .. deprecated:: 0.20
+        The `GraphLasso` was renamed to `GraphicalLasso` in version 0.20
+        and will be removed in 0.22.
+
+    Read more in the :ref:`User Guide <sparse_inverse_covariance>`.
+
+    Parameters
+    ----------
+    alpha : positive float, default 0.01
+        The regularization parameter: the higher alpha, the more
+        regularization, the sparser the inverse covariance.
+
+    mode : {'cd', 'lars'}, default 'cd'
+        The Lasso solver to use: coordinate descent or LARS. Use LARS for
+        very sparse underlying graphs, where p > n. Elsewhere prefer cd
+        which is more numerically stable.
+
+    tol : positive float, default 1e-4
+        The tolerance to declare convergence: if the dual gap goes below
+        this value, iterations are stopped.
+
+    enet_tol : positive float, optional
+        The tolerance for the elastic net solver used to calculate the descent
+        direction. This parameter controls the accuracy of the search direction
+        for a given column update, not of the overall parameter estimate. Only
+        used for mode='cd'.
+
+    max_iter : integer, default 100
+        The maximum number of iterations.
+
+    verbose : boolean, default False
+        If verbose is True, the objective function and dual gap are
+        plotted at each iteration.
+
+    assume_centered : boolean, default False
+        If True, data are not centered before computation.
+        Useful when working with data whose mean is almost, but not exactly
+        zero.
+        If False, data are centered before computation.
+
+    Attributes
+    ----------
+    covariance_ : array-like, shape (n_features, n_features)
+        Estimated covariance matrix
+
+    precision_ : array-like, shape (n_features, n_features)
+        Estimated pseudo inverse matrix.
+
+    n_iter_ : int
+        Number of iterations run.
+
+    See Also
+    --------
+    graph_lasso, GraphLassoCV
+    """
+
+
+@deprecated("The 'GraphLassoCV' was renamed to 'GraphicalLassoCV' "
+            "in version 0.20 and will be removed in 0.22.")
+class GraphLassoCV(GraphicalLassoCV):
```

Comment:
can we just assign the docs or does sphinx not like that?

Reply:
Yes, we can. Sphinx uses `.. deprecated::` tag in docstring for deprecation messages in docs

Do I need to remove deprecation decorator here or in `GraphLasso` and `graph_lasso` too?

Reply:
I meant so as not to repeat the docstring, not removing the decorator.

Reply:
I removed deprecation tag from docstrings. Looks like it is ok according to [docs](http://scikit-learn.org/stable/developers/contributing.html#deprecation)